src/pages/en/guides/data-fetching.md

```diff
@@ -4,51 +4,125 @@ title: Data Fetching
+All [Astro components](/en/core-concepts/astro-components) have access to the [global `fetch()` function](https://developer.mozilla.org/en-US/docs/Web/API/fetch) in their component script to make HTTP requests to APIs. This fetch call will be executed at build time, and the data will be available to the component template for generating dynamic HTML. 
 
-Even though Astro component scripts run inside of Node.js (and not in the browser) Astro provides this native API so that you can fetch data at page build time.
+💡 [**Top-level await**](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await#top_level_await) is supported in your Astro component script.
+
+💡 Fetched data can be passed to both Astro and framework components, as props.
+
+```astro
+// src/components/User.astro
+---
+import Contact from '../components/Contact.jsx';
+import Location from '../components/Location.astro';
+
+const response = await fetch('https://randomuser.me/api/');
+const data = await response.json();
+const randomUser = data.results[0]
+---
+<!-- Data fetched at build can be rendered in HTML -->
+<h1>User</h1>
+<h2>{randomUser.name.first} {randomUser.name.last}</h2>
+
+<!-- Data fetched at build can be passed to components as props -->
+<Contact client:load email={randomUser.email} />
+<Location city={randomUser.location.city} />
+```
+
+### requestOptions
+
+You can specify any necessary request options for your API fetch call in your Astro page component script.
 
 ```astro
 ---
-// Movies.astro
-const response = await fetch('https://example.com/movies.json');
+import MyReactApp from '../components/MyReactApp.jsx';
+
+const headers = new Headers()
+headers.append("API_KEY", import.meta.env.API_KEY)
+
+const requestOptions = {
+  method: "GET",
+  headers: headers,
+  redirect: "follow",
+  cors:"no-cors"
+};
+
+const response = await fetch(
+  `https://api.ebird.org/v2/data/obs/CA-PE/recent/notable?detail=full`,
+  requestOptions
+);
 const data = await response.json();
-// Remember: Astro component scripts log to the CLI
 console.log(data);
 ---
-<!-- Output the result to the page -->
-<div>{JSON.stringify(data)}</div>
+  <MyReactApp dataForApp={data}/>
 ```
 
-## Top-level await
+### GraphQL queries
 
-`await` is another native JavaScript feature that lets you await the response of some asynchronous promise ([MDN<span class="sr-only">- await</span>](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await)). Astro supports `await` in the top-level of your component script.
+Astro can also use `fetch()` to query a GraphQL server at build time with any valid GraphQL query. 
 
-**Important:** These are not yet available inside of non-page Astro components. Instead, do all of your data loading inside of your pages, and then pass them to your components as props.
+```astro
+---
+import BaseLayout from '../../layouts/BaseLayout.astro';
```

Comment:
I'd remove the usage of `BaseLayout` from the example, that's the one thing blocking this from being copy-pastable into some user's existing project.